### PR TITLE
Do not revert step by step to draft under certain circumstances

### PR DESCRIPTION
**Pull request by brucebolt**
Sometimes a reviewer or review requester will want to make minor changes without getting stuck in a 2i review loop.  This change prevents the status from changing to 'draft' when a step by step that is currently submitted for 2i or in 2i review is edited.  If a step by step has been approved, published or scheduled, then the status is returned to 'draft' and another 2i review is required.

Trello card: https://trello.com/c/4FN9RaqT

### CODE REVIEW

File: spec/services/step_nav_publisher_spec.rb [hunk 1]
```diff
@@ -29,6 +31,40 @@
         expect(Services.publishing_api).to have_received(:put_content).with(content_id, payload)
       end
     end
+
+    context "marks draft updated" do
```
Comment (ChrisBAshton) [inline spec/services/step_nav_publisher_spec.rb]: Kevin has picked up on this in one of my PRs, but our use of `context` is quite confusing. I'd expect this to be a `describe "whether a step by step should be reverted to draft when edited"`, _or_ a `context "there is a step by step page with steps"`.

But it looks like `context` is used in this way throughout the file, so probably one to tackle in a future cleanup PR 🤔

File: spec/services/step_nav_publisher_spec.rb [hunk 2]
```diff
@@ -29,6 +31,40 @@
...
+      let(:step_nav) { create(:step_by_step_page_with_steps) }
+
+      it "does not mark submitted for 2i review as draft" do
```
Comment (ChrisBAshton) [inline spec/services/step_nav_publisher_spec.rb]: I'd rephrase all of these to be less dev-speak:

`it "does not revert to draft if a step by step in `submitted_for_2i` state is edited"`

File: spec/services/step_nav_publisher_spec.rb [hunk 3]
```diff
@@ -29,6 +31,40 @@
...
+        step_nav.update_attributes(:status => "submitted_for_2i", :review_requester_id => review_requester_user.uid, :reviewer_id => reviewer_user.uid)
```
Comment (ChrisBAshton) [inline spec/services/step_nav_publisher_spec.rb]: Any reason why we're not using `status: "submitted_for_2i"` syntax rather than the rocket?

File: spec/services/step_nav_publisher_spec.rb [hunk 4]
```diff
@@ -29,6 +31,40 @@
...
+        expect(step_nav.status).to eq "submitted_for_2i"
+      end
+
+      it "does not mark in 2i review as draft" do
+        step_nav.update_attributes(:status => "in_review", :review_requester_id => review_requester_user.uid, :reviewer_id => reviewer_user.uid)
+        StepNavPublisher.update(step_nav)
+        expect(step_nav.status).to eq "in_review"
+      end
+
+      it "marks 2i approved as draft" do
```
Comment (ChrisBAshton) [inline spec/services/step_nav_publisher_spec.rb]: Again, think this would be clearer as `it "should revert to draft if a step by step in 'approved_2i' state is edited"`.

File: spec/services/step_nav_publisher_spec.rb (hunk 3 above)
Comment (brucebolt) [reply spec/services/step_nav_publisher_spec.rb]: No particular reason.  The usage appears inconsistent throughout the Collections Publisher tests (e.g. `spec/controllers/review_controller_spec.rb` uses rockets for `update_attributes` whereas `spec/factories.rb` uses the colon).

Comment (ChrisBAshton) [reply spec/services/step_nav_publisher_spec.rb]: Hmm and there's a mixture in this file too. I think generally we prefer the colon syntax unless there's a reason to use the rocket syntax (such as the name of the symbol being incompatible with colon syntax).
I'd be tempted to use the colon syntax here to try to nudge the codebase in the right direction (a cleanup PR later will probably still be needed on the repo as a whole!). It's also a bit less overhead to read.

Comment (brucebolt) [reply spec/services/step_nav_publisher_spec.rb]: I've updated these lines to use the colon syntax
